setup.py

```diff
@@ -81,36 +82,29 @@ def package_files(directory: str):
     'sphinxcontrib.katex>=0.8.6',
     'sphinxext.opengraph>=0.4.2',
     'sphinxemoji>=0.2.0',
-    'sphinx_rtd_theme>=1.0.0',
     'furo>=2022.1.2',
     'sphinx-copybutton>=0.4.0',
-    'autodocsumm>=0.2.7',
     'testbook>=0.4.2',
     'myst-parser>=0.15.2',
     'pylint>=2.12.2',
     'docformatter>=1.4',
 ]
 
-extra_deps['logging'] = ['wandb>=0.12.2', 'apache-libcloud>=3.4.1']
-
-extra_deps['perf'] = ['torch-tb-profiler>=0.3.1', 'psutil>=5.8.0', 'tensorboard>=2.7.0']
-
-extra_deps['nlp'] = [
+extra_deps['extra'] = [
+    # All conditional imports should go here
+    'deepspeed>=0.5.5',
+    'wandb>=0.12.2',
+    'apache-libcloud>=3.4.1',
+    'torch-tb-profiler>=0.3.1',
+    'psutil>=5.8.0',
```

Comment:
I'm thinking `torch-tb-profiler` and `psutil` should go in mosaicml[dev].  Profiling implies dev work and these modules are relatively light.  We can leave `tensorboard` as an `extra` depepndency.

Reply:
`dev` should only be used for those developing composer, since it includes a lot of extra bloat for sphinx and docs. I wouldn't want someone who is debugging performance to have to install those dependencies.

I would be OK with moving `torch-tb-profiler` and `psutil` to core, if they should always be installed. Thoughts?

Reply:
Yes in that light moving to `core` makes sense to me.

Reply:
Moving `psutil` to core. I realize that none of our code depends on `torch-tb-profiler` or `tensorboard`, so going to leave those out of the requirements.